### ISSUE TITLE
sdf c reader cannot be compiled with Xcode c99 on MacOS Big Sur

### ISSUE THREAD

**Issue by ldilucchio** [epochpic/epoch#273]
Dear Sirs ,
I am having an issue with the SDF package C compiler ( and also Fortran) on a MacOS BIgSur. The Xcode c99  gives the following error:
../SDF_lib/sdf_derived.c:44:23: error: implicit declaration of function 'vector_new' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
    vector_t *faces = vector_new();
 and than a cascade of errors.
I am desperately trying to set the build options for Xcode to C11, because it may help I suppose, but it looks like it is more difficult than foreseen.
Is there any other option, apart from switching to an Ubuntu or Windows machine?
Best regards
Laura Di Lucchio

**Comment by TomGoffrey** [comment]
Well, you could simply disable the warning (which then causes an error due to `-Werror`) but I'm a bit perplexed as to why you're seeing it in the first place. I'm also confused by the line number reported by the compiler. Do you have an up to date version of SDF?

**Comment by ldilucchio** [comment]
Yes, it is the latest version of SDF modules, updated from GitHub. So far it does not seem to work  when I try and switch off the warnings, the cascade goes on.

**Comment by keithbennett** [comment]
I'm fairly confused as to what is going on here. Firstly, there is no need to use the Xcode c99 compiler on Big Sur. The default compiler is clang version 12.0.5 and there is a "gcc" alias to this in the default setup, so there should be no need to change anything. On my machine, the code compiles without any warning or error. Secondly, the code is located in "SDF/C/src" so I'm not sure where the "../SDF_lib/sdf_derived.c" error came from. Thirdly, the line which seems to be causing the error on your machine is located on line 155 in the current version of the code. However, the error message you have posted refers to line 44.

**Comment by ldilucchio** [comment]
Unfortunately, after checking again and re-installing the sdf C folder, by doing make I get the following message again:

src/sdf_extension_util.c:174:9: error: implicit declaration of function 'sdf_extension_destroy' is invalid in C99 [-Werror,-Wimplicit-function-declaration]

...and then the errors. I will try and work on clang . Best regards
Laura Di Lucchio

**Comment by keithbennett** [comment]
I don't know how you are trying to build this. The supported method is by using the included Makefile using the unix "make" utility. However, this does not use the flags "-Werror" or "-Wimplicit-function-declaration". If I add those to the Makefile then I still do not see the errors that you report. Indeed, I can't see why they would be issued since "sdf_extension_destroy" is declared just 12 lines above (line 162) within the same function call.

**Comment by ldilucchio** [comment]
I absolutely agree on the necessity of putting an attention on the default options for make, and that's what I tried. Of course other options would have enabled warnings, but I did not use them. After carefully checking that, I am trying to re-read the clang manual in order to look for options and/or #pragma instructions that could just disable the diagnostics. So far I cannot tell either of which could work, because in principle the settings are fine.

**Comment by ldilucchio** [comment]
OF course it was the clang compiler. Now, after changing a few things, it all works fine, doing make and building after. Thanks.
Best regards
Laura Di Lucchio

**Comment by ldilucchio** [comment]
Dear sirs and madams,
I am now getting an error with python reading sdf like
Traceback (most recent call last):
  File "cep-exey.py", line 11, in <module>
    import sdf
ImportError: dynamic module does not define init function (initsdf)
.Of course I have checked my installation of epoch, SDF, FORTRAN and python. However, I cannot download the new SDF package version from the Github. Is there a way to receive the material as a full package and not a zip file? I have a Macbook MacOS

**Comment by TomGoffrey** [comment]
This is an old issue, but you can now install a standalone sdf reader via pip: https://pypi.org/project/sdfr/

If you encounter issues with this new package, please open a new issue.
